pkg/stanza/docs/operators/file_input.md

```diff
@@ -11,6 +11,7 @@ The `file_input` operator reads logs from files. It will place the lines read in
 | `include`                       | required         | A list of file glob patterns that match the file paths to be read. |
 | `exclude`                       | []               | A list of file glob patterns to exclude from reading. |
 | `poll_interval`                 | 200ms            | The duration between filesystem polls. |
+| `poll_file_limit`               | 0                | A limit on the number of files that can be consumed for a single polling cycle. A value of 0 implies an unlimited number of files. The value must be greater than or equal to `max_concurrent_files`. Ideally the value should be a multiple of `max_concurrent_file` or else the limit may be exceeded before it is enforced. |
```

Comment:
I got halfway through the review before it really clicked for me that this setting is only used when `max_concurrent_files` is enforced. This makes me think the meaning and description should very clearly follow from the explanation of `max_concurrent_files`.

What do you think about changing this to `max_batches`, with a description like "Only applicable when files must be batched in order to respect `max_concurrent_files`. This value limits the number of batches that will be processed during a single poll interval. A value of 0 indicates no limit."

This does a couple things for us:
1. I think the explanation is clearer because it directly explains why and how the settings are related.
2. We don't need to have a rule about how one value must be greater or equal to the other.
3. The implementation remains simple, and functionality is the same as if they follow the recommendation of using a multiple of `max_concurrent_files`.
4. We can strictly adhere to the max without any complicated logic.

WDYT?

Reply:
Yeah, I think that makes sense. I think the only issue is that batches is a hidden concept it's really `max_concurrent_files`/2. I don't know if anyone is going to math it out but If I have a `max_concurrent_files` of 20 and a `max_batches` of 1 I'll only see 10 files read and it might not be clear why.

Reply:
That's true, but we do mention batching in the description of `max_concurrent_files`. I think we could reword it a bit to introduce batches in a more formal way.

Reply:
Ok, I don't think it's critical to this change more just wanted to point it out. It's something we can address down the road.

Reply:
@djaglowski made the requested changes.